chainer/functions/array/get_item.py

```diff
@@ -13,16 +16,26 @@ class GetItem(function.Function):
     def __init__(self, slices):
         if not isinstance(slices, collections.Iterable):
             slices = tuple([slices])
+        n_ellipses = 0
+        for s in slices:
+            if numpy.isscalar(s) or s is None or isinstance(s, slice):
+                pass
+            elif s is Ellipsis:
+                n_ellipses += 1
+            else:
+                raise ValueError('Only basic indexing is supported')
+        if n_ellipses > 1:
+            raise ValueError('Only one Ellipsis is allowed')
```

Comment:
I think these type checks should be disabled in non-debug mode, because it contains many type tests. It would be better to wrap it by `if chainer.debug` or move it to `check_type_forward` (the latter might be intuitive).